### ISSUE TITLE
Add option to restart Mopidy service

### ISSUE THREAD

**Issue by jcass77** [pimusicbox/mopidy-websettings#17]
It may be useful to have an option available to restart the Mopidy service to deal with situations where the actor dies unexpectedly (e.g. due to a misbehaving backend extension).

This would be quicker than having to do a full reboot, and easier than having to log in to the terminal to issue the command manually.

It probably means just adding ```WebServiceRestartRequestHandler``` that does ```service mopidy restart```, similar to https://github.com/pimusicbox/mopidy-websettings/blob/1c55d944364775b675827a3a7432974e42459298/mopidy_websettings/__init__.py#L143-L151?

**Comment by kingosticks** [comment]
I think when we played around before with having it restart itself it
didn't work. Presumably because when you stop you obviously then can't
start again! So you need to do the restart from another process. I think.

**Comment by kingosticks** [comment]
And in musicbox we have a process manager to restart mopidy (and other
things) if/when they crash. The name of it escapes me.

**Comment by jcass77** [comment]
Yes you may be thinking of [monit](https://github.com/pimusicbox/pimusicbox/issues/285).

The problem is that the Mopidy actor may be dead even though Mopidy itself keeps on responding on port 6680 (which is what monit checks for). This means that if an internal Mopidy error occurs that causes certain extensions to stop working monit will not always pick up on it.

I haven't been able to figure out how to get the service to restart itself though. Simply spawning a separate process to run the os command does not appear to be sufficient as the SIGTERM signal is propagated to all subprocesses, which means they all die when the main process goes down.

**Comment by kingosticks** [comment]
That is a good point. 

You can probably send a command to monit and have it do the restart but that's not very helpful if you don't have monit running. 

> Simply spawning a separate process to run the os command does not appear to be sufficient as the SIGTERM signal is propagated to all subprocesses, which means they all die when the main process goes down.

I think it needs to be a different process group to stop this happening. Or maybe just use python-daemon?

**Comment by jcass77** [comment]
I'm not getting anywhere with this.  I tried creating a temporary shell script that ignores SIGTERM and creates a subprocess from within the script:
```
script = '''#!/bin/sh
python -c 'import subprocess, signal\nsignal.signal(signal.SIGTERM, signal.SIG_IGN)\nsubprocess.Popen(["service", "mopidy", "restart"])'
'''
script_file = tempfile.NamedTemporaryFile('wt')
script_file.write(script)
script_file.flush()

subprocess.call(["sh", script_file.name], close_fds=True)
```

This, and everything else I've tried, just dies along with the main mopidy process.

Messing with the process groups and dealing with sessions and orphans seems pretty complicated (http://linux.die.net/man/2/setpgid). 

Does anybody have any code examples that we could perhaps use as a baseline?
